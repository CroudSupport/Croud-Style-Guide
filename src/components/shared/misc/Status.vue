<template>
    <div :class="classes">
        {{ status }}
    </div>
</template>

<script>
    /**
     * Status label for tasks and jobs
     *
     * @example ./croud-status.md
     */
    export default {
        name: 'croud-status',

        props: {
            /**
             * Job or task object
             */
            context: {
                type: Object,
                required: true,
            },
        },

        computed: {
            classes() {
                return ['ui', 'label', 'circular', 'basic', 'status-label', 'fluid', 'tiny', this.context.contextual_status.colour]
            },

            status() {
                return this.context.contextual_status.status.toUpperCase()
            },
        },
    }
</script>
