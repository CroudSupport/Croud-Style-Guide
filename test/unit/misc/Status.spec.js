import Vue from 'vue'
import Status from '../../../src/components/shared/misc/Status'

const Constructor = Vue.extend(Status)

const context = {
    contextual_status: {
        status: 'in progress',
        colour: 'yellow',
    },
}

const vm = new Constructor({
    propsData: {
        context,
    },
}).$mount()

describe('Status', () => {
    it('should match the snapshot', () => {
        expect(vm.$el).toMatchSnapshot()
    })

    it('should add colour to the class', () => {
        expect(vm.classes).toContain(context.contextual_status.colour)
    })
})
