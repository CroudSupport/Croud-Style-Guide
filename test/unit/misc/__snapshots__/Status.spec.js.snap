// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status should match the snapshot 1`] = `
<div
  class="ui label circular basic status-label fluid tiny yellow"
>
  IN PROGRESS
</div>
`;
